Update tyk-stack README.md (#152)
TykTechnologies/tyk-charts: tyk-stack/README.md

## Tyk Stack

> `tyk-stack` provides the default deployment of Tyk Self Managed on a cluster. It will deploy all required Tyk components with the settings provided in the values.yaml file.

Reviewer: Same as https://github.com/TykTechnologies/tyk-charts/pull/152/files#r1377226779. If it makes sense for you, we can update them. Otherwise, please ignore my comment.
```suggestion
`tyk-stack` provides the default deployment of Tyk Self Managed on a cluster. It will deploy all required Tyk components with the settings provided in the `values.yaml` file.
```

helm show values tyk-helm/tyk-stack > values-stack.yaml --devel
```

> At a minimum, modify values-stack.yaml for the following settings:

Reviewer: Since other references to `values-stack.yaml` is quoted, I think it would be better to quote this one too
```suggestion
At a minimum, modify `values-stack.yaml `for the following settings:
```

> At a minimum, modify values-tyk-stack.yaml for the following settings:

Reviewer: ```suggestion
At a minimum, modify values-stack.yaml for the following settings:
```

1. [Set Redis connection details](#set-redis-connection-details-required)
2. [Set Mongo or PostgresSQL connection details](#set-mongo-or-postgressql-connection-details-required)

> 3. [Dashboard License](#dashboard-license)

Reviewer: Additionally, would it be good to include an additional conditional step for the Developer Portal license, in case users opt to use it?

Author: 👍 will add to it

helm upgrade tyk-stack tyk-helm/tyk-stack -n tyk --devel
```

_Note: Upgrading from tyk-pro chart_

If you were using `tyk-pro` chart for existing release, you cannot upgrade directly. Please modify the values.yaml base on your requirements and install using the new `tyk-stack` chart.

## Configuration

To get all configurable options with detailed comments:

```bash

> helm show values tyk-helm/tyk-stack > values-tyk-stack.yaml --devel

Reviewer: Should it be `values-tyk-stack` or `values-stack`?

Author: let's use values-stack.yaml

Author: ```suggestion
helm show values tyk-helm/tyk-stack > values-stack.yaml --devel
```

```

> You can update any value in your local `values.yaml` file and use `-f [filename]` flag to override default values during installation. 

Author: ```suggestion
You can update any value in your local `values-stack.yaml` file and use `-f [filename]` flag to override default values during installation. 
```

> To explore the list of supported backends for Tyk Pump, please visit https://tyk.io/docs/tyk-stack/tyk-pump/other-data-stores/.

#### Prometheus Pump

> Add `prometheus` to `pump.backend`, and add connection details for prometheus under `pump.prometheusPump`. 

Reviewer: ```suggestion
Add `prometheus` to `tyk-pump.pump.backend`, and add connection details for Prometheus under `tyk-pump.pump.prometheusPump`. 
```

> We also support monitoring using Prometheus Operator. All you have to do is set `pump.prometheusPump.prometheusOperator.enabled` to true.

Reviewer: ```suggestion
We also support monitoring using Prometheus Operator. All you have to do is set `tyk-pump.pump.prometheusPump.prometheusOperator.enabled` to true.
```

<!-- END import from pump doc -->

#### Tyk Dashboard

> The Tyk Dashboard can be configured by modifying the values under "tyk-dashboard" section of the values.yaml file

Reviewer: ```suggestion
The Tyk Dashboard can be configured by modifying the values under "tyk-dashboard" section of the `values.yaml` file
```

The chart is provided with sane defaults such that the only hard requirement is the license which needs to be put under

> .Values.global.license.dashboard in order for the bootstrapping process to work.

Reviewer: ```suggestion
`.Values.global.license.dashboard` in order for the bootstrapping process to work.
```

      ##   - name: ca-certs
      ##     mountPath: /etc/ssl/certs/ca-certs.crt
      ##     readOnly: true
      extraVolumeMounts: []
      mounts: []
  
      # Dashboard will only bootstrap if the master bootstrap option is set to true.
      bootstrap: true
  
      # The hostname to bind the Dashboard to.
      hostName: tyk-dashboard.local
      # If set to true the Dashboard will use SSL connection.
      # You will also need to set the:
      # - TYK_DB_SERVEROPTIONS_CERTIFICATE_CERTFILE
      # - TYK_DB_SERVEROPTIONS_CERTIFICATE_KEYFILE
      # variables in extraEnvs object array to define your SSL cert and key files.
      tls: false
  
      # Dashboard admin information.
      adminUser:
        firstName: admin
        lastName: user
        email: default@example.com
        # Set a password or a random one will be assigned.
        password: "123456"
      # Dashboard Organisation information.
```
### Tyk Developer Enterprise Portal Configurations

To enable Tyk Developer Enterprise Portal, set `global.components.devPortal` to true, and configure below inside `tyk-dev-portal` section.

#### Tyk Developer Enterprise Portal License (Required)

> Tyk Developer Enterprise Portal License is required. It can be set up in `tyk-dev-portal.license` or through secret `global.secrets.useSecretName`. The secret should contain a key called DeveloperPortalLicense.

Reviewer: ```suggestion
Tyk Developer Enterprise Portal License is required. It can be set up in `tyk-dev-portal.license` or through secret `global.secrets.useSecretName`. The secret should contain a key called `DevPortalLicense`.
```

```yaml
tyk-dev-portal:
  # Tyk Developer Portal license.
  license: ""
```

#### Storage Settings

> Tyk Enterprise Portal supports different storage options for storing the portal's CMS assets such as images, theme files and Open API Specification files. Please see the [Enterprise Portal Storage settings]({{<ref "tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration#portal-settings">}}) page for all the available options. Helm chart supports the setting of following fields in values.yaml `storage` section:

Reviewer: Same as https://github.com/TykTechnologies/tyk-charts/pull/152/files#r1376606723

Author: ```suggestion
Tyk Enterprise Portal supports different storage options for storing the portal's CMS assets such as images, theme files and Open API Specification files. Please see the [Enterprise Portal Storage settings]({{<ref "tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration#portal-settings">}}) page for all the available options. Helm chart supports the setting of following fields in values-stack.yaml `storage` section:
```

Author: ```suggestion
Tyk Enterprise Portal supports different storage options for storing the portal's CMS assets such as images, theme files and Open API Specification files. Please see the [Enterprise Portal Storage settings](https://tyk.io/docs/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration#portal-settings) page for all the available options. Helm chart supports the setting of following fields in values.yaml `storage` section:
```

> Tyk Enterprise Portal supports different storage options for storing the portal's CMS assets such as images, theme files and Open API Specification files. Please see the [Enterprise Portal Storage settings](https://tyk.io/docs/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration#portal-settings) page for all the available options. Helm chart supports the setting of following fields in values.yaml `storage` section:

Reviewer: ```suggestion
Tyk Enterprise Portal supports different storage options for storing the portal's CMS assets such as images, theme files and Open API Specification files. Please see the [Enterprise Portal Storage settings](https://tyk.io/docs/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration#portal-settings) page for all the available options. Helm chart supports the setting of the following fields in `values.yaml` `storage` section:
```

    # Supported types: fs, s3, db
    type: "fs"
    database:
      dialect: "sqlite3"
      connectionString: "db/portal.db"
      enableLogs: false
      maxRetries: 3
      retryDelay: 5000
    # Configuration values for using s3 as storage for Tyk Developer Portal
    # In case you want to provide the key ID and access key via secrets please
    # refer to the existing secret inside the helmchart or the
    # .Values.global.secrets.useSecretName variable and a secret containing
    # the keys DeveloperPortalAwsAccessKeyId and respectively,
    # DeveloperPortalAwsSecretAccessKey
    s3:
      awsAccessKeyid: your-access-key
      awsSecretAccessKey: your-secret-key
      region: sa-east-1
      endpoint: your-portal-bucket
      bucket: https://s3.sa-east-1.amazonaws.com
      acl: private
      presign_urls: true
    persistence:
      # User can mount existing PVC to the Tyk Developer Portal
      # Make sure to change the kind to Deployment if you are mounting existing PVC 
      mountExistingPVC: ""
      storageClass: ""
      accessModes:
        - ReadWriteOnce
      size: 8Gi
      annotations: {}
      labels: {}
      selector: {}
```

#### Other Configurations

> Other [Enterprise Portal configurations]({{<ref "tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration">}}) can be set by using environment variables with `extraEnvs` fields, e.g.:

Reviewer: This linking `{{<ref "tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration">}}` won't work in GitHub readme.

Author: ```suggestion
Other [Enterprise Portal configurations](https://tyk.io/docs/tyk-stack/tyk-developer-portal/enterprise-developer-portal/install-tyk-enterprise-portal/configuration) can be set by using environment variables with `extraEnvs` fields, e.g.:
```
